packages/flutter/lib/src/cupertino/page.dart

```diff
@@ -25,20 +25,113 @@ final FractionalOffsetTween _kBottomUpTween = new FractionalOffsetTween(
-final DecorationTween _kShadowTween = new DecorationTween(
-  begin: BoxDecoration.none, // No shadow initially.
-  end: const BoxDecoration(
-    boxShadow: const <BoxShadow>[
-      const BoxShadow(
-        blurRadius: 10.0,
-        spreadRadius: 4.0,
-        color: const Color(0x38000000),
-      ),
-    ],
+// BoxDecoration from no shadow to page shadow mimicking iOS page transitions
+// using gradients.
+final DecorationTween _kGradientShadowTween = new DecorationTween(
+  begin: _CupertinoEdgeShadowDecoration.none,
+  end: const _CupertinoEdgeShadowDecoration(
+    edgeGradient: const LinearGradient(
+      // Spans 5% of the page.
+      begin: const FractionalOffset(0.95, 0.0),
+      end: FractionalOffset.topRight,
+      colors: const <Color>[
+        const Color(0x00000000), 
+        const Color(0x04000000),
+        const Color(0x12000000),
+        const Color(0x38000000)
+      ],
+      stops: const <double>[0.0, 0.3, 0.6, 1.0],
+    ), 
   ),
 );
 
+/// A custom [Decoration] used to paint an extra shadow on the left edge of the 
+/// box it's decorating. It's like a [BoxDecoration] with only a gradient except 
+/// it paints to the left of the box instead of behind the box.
+class _CupertinoEdgeShadowDecoration extends Decoration {
+  const _CupertinoEdgeShadowDecoration({ this.edgeGradient });
+
+  static const _CupertinoEdgeShadowDecoration none = 
+      const _CupertinoEdgeShadowDecoration();
+
+  /// A gradient to draw to the left of the box being decorated. 
+  /// FractionalOffsets are relative to the original box translated one box
+  /// width to the left.
+  final LinearGradient edgeGradient;
+
+  /// Linearly interpolate between two edge shadow decorations decorations.
+  ///
+  /// See also [Decoration.lerp].
+  static _CupertinoEdgeShadowDecoration lerp(
+    _CupertinoEdgeShadowDecoration a, 
+    _CupertinoEdgeShadowDecoration b, 
+    double t
+  ) {
+    if (a == null && b == null)
+      return null;
+    return new _CupertinoEdgeShadowDecoration(
+      edgeGradient: LinearGradient.lerp(a?.edgeGradient, b?.edgeGradient, t),
+    );
+  }
+
+  @override
+  _CupertinoEdgeShadowDecoration lerpFrom(Decoration a, double t) {
+    if (a is! _CupertinoEdgeShadowDecoration)
+      return _CupertinoEdgeShadowDecoration.lerp(null, this, t);
+    return _CupertinoEdgeShadowDecoration.lerp(a, this, t);
+  }
+
+  @override
+  _CupertinoEdgeShadowDecoration lerpTo(Decoration b, double t) {
+    if (b is! _CupertinoEdgeShadowDecoration)
+      return _CupertinoEdgeShadowDecoration.lerp(this, null, t);
+    return _CupertinoEdgeShadowDecoration.lerp(this, b, t);
+  }
+  
+  @override
+  _CupertinoEdgeShadowPainter createBoxPainter([VoidCallback onChanged]) {
+    return new _CupertinoEdgeShadowPainter(this, onChanged);
+  }
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
+    if (other is! _CupertinoEdgeShadowDecoration)
```

Comment:
compare runtimeType rather than using is, so that instances of subclasses aren't == to instances of superclasses